src/js/game/LevelMVC/LevelView.js

```diff
@@ -1903,6 +1904,31 @@ module.exports = class LevelView {
         sprite.animations.add("close", frameList, 5, false);
         break;
 
+      case "doorIron":
+        atlas = this.blocks[blockType][0];
+        frame = this.blocks[blockType][1];
+        xOffset = this.blocks[blockType][2];
+        yOffset = this.blocks[blockType][3];
+        sprite = plane.create(xOffset + 40 * x, yOffset + plane.yOffset + 40 * y, atlas, frame);
+
+        frameList = [];
+        var animationFramesIron = Phaser.Animation.generateFrameNames("DoorIron", 0, 3, "", 1);
+        for (let j = 0; j < 5; ++j) {
+          frameList.push("DoorIron0");
+        }
+        frameList = frameList.concat(animationFramesIron);
+
+        sprite.animations.add("open", frameList, 5, false);
+
+        frameList = [];
+        animationFramesIron = Phaser.Animation.generateFrameNames("DoorIron", 3, 0, "", 1);
+        for (let j = 0; j < 5; ++j) {
+          frameList.push("DoorIron3");
+        }
+        frameList = frameList.concat(animationFramesIron);
+        sprite.animations.add("close", frameList, 5, false);
+        break;
+
```

Comment:
Can we de-dupe this code with `case "door"` above?

Reply:
The duping was mainly because all the sprite frame references are different. Door0 vs DoorIron0. 

I could take that block out into a helper function and pass in 'Door' vs 'IronDoor' and evaluate it before the switch statement? Let me know what you think the best way to go about that is.

Reply:
I'm thinking something along the lines of:

```js
const buildDoor = (levelView, type) => {
  atlas = levelView.blocks[blockType][0];
  frame = levelView.blocks[blockType][1];
  xOffset = levelView.blocks[blockType][2];
  yOffset = levelView.blocks[blockType][3];
  sprite = plane.create(xOffset + 40 * x, yOffset + plane.yOffset + 40 * y, atlas, frame);

  frameList = [];
  var animationFramesIron = Phaser.Animation.generateFrameNames(type, 0, 3, "", 1);
  for (let j = 0; j < 5; ++j) {
    frameList.push(`${type}0`);
  }
  frameList = frameList.concat(animationFramesIron);

  sprite.animations.add("open", frameList, 5, false);

  frameList = [];
  animationFramesIron = Phaser.Animation.generateFrameNames(type, 3, 0, "", 1);
  for (let j = 0; j < 5; ++j) {
    frameList.push(`${type}3`);
  }
  frameList = frameList.concat(animationFramesIron);
  sprite.animations.add("close", frameList, 5, false);

  return sprite;
};

// ...

case "door":
  sprite = buildDoor(this, "Door");
  break;

case "doorIron":
  sprite = buildDoor(this, "IronDoor");
  break;
```

Reply:
Roger that. 👍

Reply:
Curious non-important question: `${type}0` vs type + "0"
Strings in js have the + operator function as a concatenation correct? Is there a reason for 1 vs the other? Or is it just readability / style?